docs/reference/mapping/runtime.asciidoc

```diff
@@ -747,6 +747,243 @@ address.
+      "type": "long",
+      "script": {
+        "source": """
+          for (double v : doc['voltage'])
+          emit((long)(v * params['multiplier']))
+          """,
+        "params": {
+          "multiplier": 2
+        }
+      }
+    }
+  }
+}
+----
+// TEST[continued]
+
+You retrieve the calculated values using the <<search-fields,`fields`>>
+parameter on the `_search` API:
+
+[source,console]
+----
+GET my-index/_search
+{
+  "fields": [
+    "voltage_reading",
+    "node"
+    ]
+}
+----
+// TEST[continued]
+
+After reviewing the sensor data and running some tests, you determine that the
+multiplier for reported sensor data should be `4`. To gain greater performance,
+you decide to index the `voltage_reading` runtime field with the new
+`multiplier` parameter.
+
+In a new index named `my-index-00001`, copy the `voltage_reading` runtime field
+definition into the mappings of the new index. It's that simple!
+
+[source,console]
+----
+PUT my-index-00001/
+{
+  "mappings": {
+    "properties": {
+      "timestamp": {
+        "type": "date"
+      },
+      "temperature": {
+        "type": "long"
+      },
+      "voltage": {
+        "type": "double"
+      },
+      "node": {
+        "type": "keyword"
+      },
+      "voltage_reading": {
+        "type": "long",
+        "script": {
+          "source": """
+            for (double v : doc['voltage'])
+            emit((long)(v * params['multiplier']))
+            """,
+          "params": {
+            "multiplier": 4
+          }
+        }
+      }
+    }
+  }
+}
+----
+
+Bulk index some sample data from your sensors into the `my-index-00001` index:
+
+[source,console]
+----
+POST my-index-00001/_bulk?refresh=true
+{ "index": {}}
+{ "timestamp": 1516729294000, "temperature": 200, "voltage": 5.2, "node": "a"}
+{ "index": {}}
+{ "timestamp": 1516642894000, "temperature": 201, "voltage": 5.8, "node": "b"}
+{ "index": {}}
+{ "timestamp": 1516556494000, "temperature": 202, "voltage": 5.1, "node": "a"}
+{ "index": {}}
+{ "timestamp": 1516470094000, "temperature": 198, "voltage": 5.6, "node": "b"}
+{ "index": {}}
+{ "timestamp": 1516383694000, "temperature": 200, "voltage": 4.2, "node": "c"}
+{ "index": {}}
+{ "timestamp": 1516297294000, "temperature": 202, "voltage": 4.0, "node": "c"}
+----
+// TEST[continued]
+
+You can now retrieve calculated values in a search query, and find documents
+based on precise values. The following range query returns all documents where
+the calculated `voltage_reading` is greater than or equal to `10`, but less
+than or equal to `16`:
+
+[source,console]
+----
+GET my-index-00001/_search
+{
+  "query": {
+    "range": {
+      "voltage_reading": {
+        "gte": 16,
+        "lte": 20,
+        "boost": 1.0
+      }
+    }
+  }
+  , "size": 2
+}
+----
+// TEST[continued]
+
+Even though `voltage_reading` isn't included in the response, the documents
+that match the range query are returned based on the calculated value of the
+included script:
```

Comment:
I might us the `fields` API to ask for `voltage_reading` here. I kind of expect that to be the "normal" way to do this now.

Reply:
That's a good callout, and a better user experience to actually see `voltage_corrected` in the response. This works for me, but let me know if there are ways to improve the request:

```
POST my-index-00001/_search
{
  "query": {
    "range": {
      "voltage_corrected": {
        "gte": 16,
        "lte": 20,
        "boost": 1.0
      }
    }
  },
  "fields": [
    "voltage_corrected",
    "node"]
}
```

Reply:
```
  "fields": ["voltage_corrected", "node"]
```

Is a little shorted, I guess. Otherwise its what I'd use.